File: leetcode/top-interview-questions-easy/arrays/7. plus_one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        shoudIncreasePrevious = False
        resultArray = []
        i = len(digits) - 1
        while i >=0:
            isLast = i == len(digits) - 1
            shouldAddOne = isLast or shouldIncreasePrevious
            current = digits[i]
            result = current + 1 if shouldAddOne else current
            digitResult = result % 10
            resultArray.insert(0, digitResult)
            shouldIncreasePrevious = result > 9
            i -= 1
        
        if shouldIncreasePrevious:
            resultArray.insert(0, 1)
        
        return resultArray

File: leetcode/top-interview-questions-easy/arrays/7. plus_one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
  var shouldIncreasePrevious = false;
  var resultArray = [];
  for (var i = digits.length - 1; i >= 0; i--) {
    var shouldAddOne = i === digits.length - 1 || shouldIncreasePrevious;
    var current = digits[i];
    var result = shouldAddOne ? current + 1 : current;
    var digitResult = result % 10;

    resultArray.unshift(digitResult);

    shouldIncreasePrevious = result > 9;
    //console.log(i, digits[i], result, digitResult, shouldIncreasePrevious, resultArray);
  }

  if (shouldIncreasePrevious) {
    resultArray.unshift(1);
  }

  return resultArray;
};
